src/modules/75-cd/components/PipelineSteps/K8sServiceSpec/ManifestSource/HelmChartManifestSource/HelmChartManifestSource.tsx

```diff
@@ -180,10 +187,19 @@ const Content = (props: ManifestSourceRenderProps): React.ReactElement => {
     label: region.name
   }))
 
-  const chartVersions = (chartVersionData?.data?.helmChartVersions || []).map((chartVersion: string) => ({
-    value: chartVersion,
-    label: chartVersion
-  }))
+  React.useEffect(() => {
+    if (chartVersionData?.data?.helmChartVersions) {
+      setChartVersions(
+        defaultTo(
+          chartVersionData.data.helmChartVersions.map((chartVersion: string) => ({
+            value: chartVersion,
+            label: chartVersion
+          })),
+          []
+        )
+      )
+    }
```

Comment:
In this, the defaultTo won't be called as null, and undefined values would block from entering the if case.

```suggestion
setChartVersions(
  defaultTo(chartVersionData?.data?.helmChartVersions , []).map((chartVersion) => ({
    value: chartVersion,
    label: chartVersion
  }))
);
```

Reply:
Are you right , thanks !
improved